modules/test/playwright/tests/workspaces/liferay-customer-workspace/tests/general/unit/projectUserPermission.spec.ts

```diff
@@ -0,0 +1,183 @@
+		accountAdministratorRole[0].id,
+		'test@liferay.com'
+	);
+});
+
+test.describe('Project User Permission', () => {
+	test('User can only see things related to User role', async ({
+		apiHelpers,
+		customerApiHelpers,
+		homePage,
+		page,
+		projectAttachmentsPage,
+		projectLiferayPaaSPage,
+		projectOverviewPage,
+		projectTeamMembersPage,
+	}) => {
+		await projectOverviewPage.goto(accountExternalReferenceCode);
+
+		const accountFlag = await customerApiHelpers.getAccountFlag(
+			accountExternalReferenceCode
+		);
+
+		if (accountFlag === undefined) {
+			await expect(
+				page.getByRole('button', {name: 'Start Project Setup'})
+			).toBeVisible();
+		}
+
+		await projectTeamMembersPage.goto(accountExternalReferenceCode);
+
+		await projectTeamMembersPage.inviteButton.click();
+
+		await projectTeamMembersPage.firstNameField.fill('testfirst');
+
+		await projectTeamMembersPage.lastNameField.fill('testlast');
+
+		userEmailAddress = getRandomString() + '@liferay.com';
+
+		await projectTeamMembersPage.emailField.fill(userEmailAddress);
+
+		await projectTeamMembersPage.roleSelect.click({force: true});
+
+		await projectTeamMembersPage.userRoleOption.click();
+
+		await projectTeamMembersPage.applyButton.click();
+
+		await projectTeamMembersPage.sendInvitationsButton.click();
+
+		await projectTeamMembersPage.goto(accountExternalReferenceCode);
+
+		await expect(page.getByText(userEmailAddress)).toBeVisible();
+
+		const userAccount =
+			await apiHelpers.headlessAdminUser.getUserAccountByEmailAddress(
+				userEmailAddress
+			);
+
+		await apiHelpers.headlessAdminUser.patchUserAccount(userAccount, {
+			password: 'test',
+		});
+
+		await customerPerformUserSwitch(page, userEmailAddress);
+
+		await expect(
+			page.getByRole('heading', {name: 'Test Account Liferay PaaS'})
+		).toBeVisible();
+
+		expect((await homePage.projectCard.all()).length).toBe(1);
+
+		await projectOverviewPage.goto(accountExternalReferenceCode);
+
+		await expect(projectOverviewPage.heading).toBeVisible();
+
+		await projectLiferayPaaSPage.goto(accountExternalReferenceCode);
+
+		await expect(projectLiferayPaaSPage.heading).toBeVisible();
+
+		if (projectLiferayPaaSPage.projectNotActivatedTag) {
+			await expect(
+				projectLiferayPaaSPage.finishActivationButton
+			).not.toBeVisible();
+		}
+
+		await expect(
+			page.getByRole('button', {name: 'Product Activation'})
+		).toBeDisabled();
+
+		await projectAttachmentsPage.goto(accountExternalReferenceCode);
+
+		await expect(projectAttachmentsPage.heading).toBeVisible();
+
+		await expect(
+			projectAttachmentsPage.attachmentIconContainer
+		).toBeVisible();
+
+		await projectTeamMembersPage.goto(accountExternalReferenceCode);
+
+		await expect(page.getByText(userEmailAddress)).toBeVisible();
+
+		await expect(projectTeamMembersPage.inviteButton).not.toBeVisible();
+
+		await expect(
+			projectTeamMembersPage.userActionColumnHeader
+		).not.toBeVisible();
+
+		await customerPerformLogout(page);
```

Comment:
i see you're logging out and back in as admin. Is this so the cleanup.after logic works correctly?

Reply:
yes, if don't it doesn't work.